src/session/default.ts

```diff
@@ -211,13 +230,45 @@ class DefaultSession implements ISession {
    * This uses the Jupyter REST API, and the response is validated.
    * The promise is fulfilled on a valid response and rejected otherwise.
    */
-  rename(path: string): Promise<void> {
+  setPath(path: string): Promise<void> {
     if (this.isDisposed) {
       return Promise.reject(new Error('Session is disposed'));
     }
-    let data = JSON.stringify({
-      notebook: { path }
-    });
+    let data = JSON.stringify({ path });
+    return this._patch(data).then(() => { return void 0; });
+  }
+
+  /**
+   * Update the session name.
+   *
+   * @param name - The new session name.
+   *
+   * #### Notes
+   * This uses the Jupyter REST API, and the response is validated.
+   * The promise is fulfilled on a valid response and rejected otherwise.
+   */
+  setName(name: string): Promise<void> {
+    if (this.isDisposed) {
+      return Promise.reject(new Error('Session is disposed'));
+    }
+    let data = JSON.stringify({ name });
+    return this._patch(data).then(() => { return void 0; });
+  }
+
+  /**
+   * Update the session type.
+   *
+   * @param type - The new session type.
+   *
+   * #### Notes
+   * This uses the Jupyter REST API, and the response is validated.
+   * The promise is fulfilled on a valid response and rejected otherwise.
+   */
+  setType(type: string): Promise<void> {
+    if (this.isDisposed) {
+      return Promise.reject(new Error('Session is disposed'));
+    }
+    let data = JSON.stringify({ type });
```

Comment:
Yep, available in ES6.   I am going to also verify that current master of services still works against notebook master (that the old api still works).